### PR TITLE
report PR number to build-tracker-service

### PR DESCRIPTION
**Pull request by raphaelbharel**

### CODE REVIEW

File: modules/build-raptor-action/src/build-raptor-action.ts [hunk 1]
```diff
@@ -17,8 +19,28 @@ interface Options {
   printPassing: boolean
 }
 
-function getEnv(envVarName: 'GITHUB_SHA' | 'CI') {
-  return process.env[envVarName] // eslint-disable-line no-process-env
+async function getLatestMainMergedPRNumber(): Promise<number> {
+  const response = await axios.get(
+    `https://api.github.com/repos/moojo-tech/antelope/pulls?state=closed&base=main&sort=updated&direction=desc`,
```
Comment (raphaelbharel) [inline modules/build-raptor-action/src/build-raptor-action.ts:24]: we could generalize this function, such that we read the ORG, REPO name, and the name of the base branch

File: modules/build-raptor-core/src/env-utils.ts
Comment (imaman) [inline modules/build-raptor-core/src/env-utils.ts:1]: keep it in modules/build-raptor/src/build-raptor-cli.ts.

File: modules/build-raptor/src/build-raptor-cli.ts [hunk 2]
```diff
@@ -67,8 +64,12 @@ async function run(options: Options) {
   logger.print(`logging to ${logFile}`)
   const isCi = getEnv('CI') === 'true'
   const commitHash = getEnv('GITHUB_SHA')
+  const latestMainPR = getEnv('GITHUB_MAIN_PR_NUM')
```
Comment (imaman) [inline modules/build-raptor/src/build-raptor-cli.ts:67]: call `getLatestMainMergedPRNumber()` to get this number.

File: modules/build-raptor-action/package.json
Comment (imaman) [inline modules/build-raptor-action/package.json:1]: I think you no longer need to change this file (b/c there are no other changes in build-raptor-action)

File: modules/build-raptor/package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
   },
   "devDependencies": {
     "@types/yargs": "^17.0.7",
-    "@vercel/ncc": "^0.33.1"
+    "@vercel/ncc": "^0.33.1",
+    "axios": "^1.4.0"
```
Comment (imaman) [inline modules/build-raptor/package.json:27]: the dependency needs to be added at the package.json of the module that uses it (and, it's not a devDependency).

File: modules/build-raptor/src/build-raptor-cli.ts [hunk 4]
```diff
@@ -40,17 +43,51 @@ interface Options {
 
 type TestEndedEvent = RepoProtocolEvent['testEnded']
 
+type EnvVarName = 'GITHUB_SHA' | 'GITHUB_REPOSITORY' | 'GITHUB_REF' | 'GITHUB_REPOSITORY_OWNER' | 'GITHUB_TOKEN' | 'CI'
+
+function getEnv(envVarName: EnvVarName) {
+  return process.env[envVarName] // eslint-disable-line no-process-env
+}
+
+async function findPRForCommit(commitHash: string): Promise<number | undefined> {
```
Comment (imaman) [inline modules/build-raptor/src/build-raptor-cli.ts:52]: please move to a separate file

File: modules/build-raptor/src/build-raptor-cli.ts [hunk 5]
```diff
@@ -67,8 +104,20 @@ async function run(options: Options) {
...
   const isCi = getEnv('CI') === 'true'
   const commitHash = getEnv('GITHUB_SHA')
+
+  let pullRequest: number | undefined
+  try {
+    if (commitHash) {
+      pullRequest = await findPRForCommit(commitHash)
+    }
+  } catch (error) {
+    logger.error(`Failed to find PR for commit ${commitHash}`, error)
```
Comment (imaman) [inline modules/build-raptor/src/build-raptor-cli.ts:114]: I prefer to fail hard on this - we probably would not notice an error in the log

File: modules/build-raptor/src/build-raptor-cli.ts [hunk 6]
```diff
@@ -81,15 +130,19 @@ async function run(options: Options) {
       return
     }
 
-    if (!commitHash) {
-      throw new Error(`missing commit hash in CI`)
+    const endpointData = { packageName: u.id, commitHash, pullRequest, casReference: resolved }
+    const parsedRequest = RegisterAssetRequest.safeParse(endpointData)
+
+    if (!parsedRequest.success) {
+      throw new Error(`Invalid request data: ${endpointData}`)
```
Comment (imaman) [inline modules/build-raptor/src/build-raptor-cli.ts:137]: doing a `.safeParse()` and then throwing an exception if it's unsuccessful is pretty much equivalent to doing a `.parse()`

File: modules/build-raptor/src/build-raptor-cli.ts (hunk 5 above)
Comment (raphaelbharel) [reply modules/build-raptor/src/build-raptor-cli.ts:114]: done

File: modules/build-raptor/package.json (hunk 3 above)
Comment (raphaelbharel) [reply modules/build-raptor/package.json:27]: done

File: modules/build-raptor-action/package.json
Comment (raphaelbharel) [reply modules/build-raptor-action/package.json:1]: done

File: modules/build-raptor/src/build-raptor-cli.ts (hunk 4 above)
Comment (raphaelbharel) [reply modules/build-raptor/src/build-raptor-cli.ts:52]: done

File: modules/build-raptor/src/endpoint-requester.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import { GithubResponseSchema } from './build-raptor-api'
+import { getEnv } from './build-raptor-cli'
+
+export async function getPRForCommit(commitHash: string): Promise<number | undefined> {
+  if (!commitHash || !commitHash.match(/^[a-f0-9]{40}$/)) {
+    throw new Error('Invalid commit hash.')
+  }
+
+  const repoOwner = getEnv('GITHUB_REPOSITORY_OWNER')
+  const repoName = getEnv('GITHUB_REPOSITORY')
```
Comment (imaman) [inline modules/build-raptor/src/endpoint-requester.ts:12]: let's pass these two values as parameters (i.e., the caller will call getEnv() and pass the values it gets from it as parameters)

File: modules/build-raptor/src/endpoint-requester.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  if (!repoOwner || !repoName) {
+    throw new Error('Required repo environment variable(s) missing or invalid.')
+  }
+
+  const response = await axios.get(
+    `https://api.github.com/repos/${repoOwner}/${repoName}/commits/${commitHash}/pulls`,
+    {
+      headers: {
+        Accept: 'application/vnd.github+json',
+        Authorization: `Bearer ${getEnv('GITHUB_TOKEN')}`,
```
Comment (imaman) [inline modules/build-raptor/src/endpoint-requester.ts:23]: ditto - caller will call `getEnv('GITHUB_TOKEN')` and pass it as param

**Review by imaman — APPROVED** [review]

File: modules/build-raptor/src/endpoint-requester.ts (hunk 8 above)
Comment (raphaelbharel) [reply modules/build-raptor/src/endpoint-requester.ts]: done

File: modules/build-raptor/src/endpoint-requester.ts (hunk 7 above)
Comment (raphaelbharel) [reply modules/build-raptor/src/endpoint-requester.ts]: done

**Pull request closed without merging**